flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBase.java

```diff
@@ -235,7 +243,8 @@ public FlinkKafkaConsumerBase(
 			Pattern topicPattern,
 			KeyedDeserializationSchema<T> deserializer,
 			long discoveryIntervalMillis,
-			boolean useMetrics) {
+			boolean useMetrics,
+			boolean checkUnavailablePartitions) {
```

Comment:
Why do we want this to be configurable? Is there any case that we would prefer to leave them untouched?